### DISCUSSION TITLE
How to add a string constant to a class?

### DISCUSSION THREAD

**Discussion by Mars-CN** [micropython/micropython#12232] [ESP32] [answered]
I know how to add constants of type integer to a class, which can be achieved through the following code:
{}MP_ROM_QSTR(MP_QSTR_INTVAL), MP_ROM_INT(100)}，
But how to add a string type constant to a class?

**Comment by jimmo** [comment, marked as answer]
Same as how you encode the keys -- use `{ MP_ROM_QSTR(MP_QSTR_STRVAL), MP_ROM_QSTR(MP_QSTR_FooBar ) }`

If the string can't be encoded into MP_QSTR_ then you'll either need to add it to qstrdefs.h or make a string object from a literal and point to that.

**Reply by jimmo** [reply]
See https://github.com/micropython/micropython/blob/master/py/modsys.c#L59 for an example of making a string object.

See https://github.com/micropython/micropython/blob/master/ports/stm32/main.c#L193 and https://github.com/micropython/micropython/blob/master/ports/stm32/qstrdefsport.h#L31 for an example of using qstrdefsport.h.
